pkg/rules/service.go

```diff
@@ -0,0 +1,196 @@
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/Masterminds/sprig/v3"
+	"github.com/pkg/errors"
+
+	"github.com/giantswarm/observability-operator/pkg/common/labels"
+	"github.com/giantswarm/observability-operator/pkg/common/tenancy"
+)
+
+const (
+	alloyRulesAppCatalog        = "giantswarm"
+	alloyRulesAppName           = "alloy-rules"
+	alloyRulesAppNamespace      = "giantswarm"
+	alloyRulesChartName         = "alloy"
+	alloyRulesDeployInNamespace = "monitoring"
+	alloyRulesConfigMapName     = "alloy-rules-config"
+)
+
+var (
+	//go:embed templates/alloy-rules.yaml.template
+	appConfig         string
+	appConfigTemplate *template.Template
+)
+
+// init initializes the template for the alloy-rules configmap.
+func init() {
+	appConfigTemplate = template.Must(template.New("alloy-rules.yaml").Funcs(sprig.FuncMap()).Parse(appConfig))
+}
+
+type Service struct {
+	Client          client.Client
+	AlloyAppVersion semver.Version
+}
+
+func (s *Service) Configure(ctx context.Context, cluster clusterv1.Cluster) error {
+	logger := log.FromContext(ctx)
+	logger.Info("configuring alloy-rules")
+
+	logger.Info("create or update alloy rules configmap")
+	err := s.createOrUpdateConfigMap(ctx, cluster)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	logger.Info("configure alloy rules app")
+	err = s.configureApp(ctx)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	logger.Info("alloy-rules is configured")
+
+	return nil
+}
+
+func (s *Service) CleanUp(ctx context.Context, cluster clusterv1.Cluster) error {
+	logger := log.FromContext(ctx)
+	logger.Info("deleting alloy-rules")
+
+	configmap := configMap()
+
+	err := s.Client.Delete(ctx, configmap)
+	if err != nil && !apierrors.IsNotFound(err) {
+		logger.Error(err, "failed to delete configmap %s", alloyRulesConfigMapName)
+		return errors.WithStack(err)
+	}
+
+	app := app()
+	err = s.Client.Delete(ctx, app)
+	if err != nil && !apierrors.IsNotFound(err) {
+		logger.Error(err, "failed to delete app %s", alloyRulesAppName)
+		return errors.WithStack(err)
+	}
+
+	logger.Info("deleted alloy-rules")
+	return nil
+}
+
+func configMap() *v1.ConfigMap {
+	return &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      alloyRulesConfigMapName,
+			Namespace: alloyRulesAppNamespace,
+			Labels:    labels.Common,
+		},
+	}
+}
+
+func app() *appv1.App {
+	labels := maps.Clone(labels.Common)
+	labels["app-operator.giantswarm.io/version"] = "0.0.0"
+	return &appv1.App{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      alloyRulesAppName,
+			Namespace: alloyRulesAppNamespace,
+			Labels:    labels,
+		},
+	}
+}
+
+func (s Service) createOrUpdateConfigMap(ctx context.Context, cluster clusterv1.Cluster) error {
```

Comment:
I see there are lot of changes related to passing this `cluster` variable around, but I actually don't see it being used anywhere, did I miss something ?

Reply:
I think it's because it's not anymore. I probably forgot to remove it

Reply:
Please remove this, that's extra changes we don't need then.

Reply:
That was my plan after your comment